include/swift/Basic/FileTypes.def

```diff
@@ -73,6 +73,13 @@ TYPE("module-trace",        ModuleTrace,               "trace.json",      "")
 TYPE("index-data",          IndexData,                 "",                "")
 TYPE("opt-record",          OptRecord,                 "opt.yaml",        "")
 
+// Overlay files declare wrapper modules, called "separately-imported overlays",
```

Comment:
Could you add entries to Lexicon.rst for separately-imported overlay, cross-import overlay and maybe even cross-import dir?

Reply:
That would be a good idea.

To respond to this from above:

> What is the reason behind not emitting a diagnostic?

Mainly staging—if you're trying to switch from one to the other and accidentally end up with both modules declaring the cross-import, should your clients all break? It would cause some pretty psychedelic name lookup behavior, though, so maybe we should emit a warning.

Reply:
Yeah name lookup is the thing I'm worried about. If it is a warning, it is still ok (we can live with that) for a temporary thing. If it creates some kind of error due to funky name lookup and we don't have that warning, debugging the situation will not be fun (esp. if you are on vacation when we get the bug report 😝).

Reply:
I'll add a warning, and make sure it makes it clear that this is the modules' fault.

Reply:
Added.